packages/angular/cli/src/commands/mcp/tools/build.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { z } from 'zod';
+import { execSync } from 'child_process';
+import { declareTool } from './tool-registry';
+
+const CONFIGURATIONS = {
+  development: {
+    args: '-c development',
+  },
+  production: {
+    args: '',
+  },
+};
+
+const BUILD_STATUSES = ['success', 'failure'] as const;
+type BuildStatus = (typeof BUILD_STATUSES)[number];
+
+const buildToolInputSchema = z.object({
+  project: z
+    .string()
+    .optional()
+    .describe(
+      'Which project to build in a monorepo context. If not provided, builds the top-level project.',
+    ),
+  configuration: z
+    .enum(Object.keys(CONFIGURATIONS) as [string, ...string[]])
+    .optional()
+    .describe('Which build configuration to use. Defaults to "development".'),
+});
+
+export type BuildToolInput = z.infer<typeof buildToolInputSchema>;
+
+const buildToolOutputSchema = z.object({
+  status: z.enum(BUILD_STATUSES).describe('Build status.'),
+  logs: z.string().describe('Output logs from `ng build`.'),
+  path: z.string().optional().describe('The output path the build project was written into.'),
+});
+
+export type BuildToolOutput = z.infer<typeof buildToolOutputSchema>;
+
+export function runBuild(input: BuildToolInput) {
+  const configurationName = input.configuration ?? 'development';
+  const configuration = CONFIGURATIONS[configurationName as keyof typeof CONFIGURATIONS];
+  const command = ['ng', 'build'];
+  if (input.project) {
+    command.push(input.project);
+  }
+  if (configuration.args) {
+    command.push(configuration.args);
+  }
+
+  let status: BuildStatus = 'success';
+  let logs = '';
+  let outputPath: string | undefined;
+
+  try {
+    logs = execSync(command.join(' ')).toString();
```

Comment:
Consider: Should we use the async promise version so as not to hang other queued tasks? Could the same MCP server receive multiple tool calls at the same time? Can multiple agents/prompts connect to the same MCP server?

Even if we can assume one blocking tool call at a time, we might have some background async work to do, so maybe it's worth being async anyways?

Also, I'll mention since it's not very intuitive: The way to do async with `child_process` is to feed it through `promisify` like this: https://github.com/dgp1130/rules_prerender/blob/d6a53dcfd96229bcf134b7b8c3d2c0e2921d6c1a/common/testing/binary.mts